### PR TITLE
aolserver: minor update, fix the build for macos-11/12 and Rosetta, add modeline

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Minor changes to the port. Update to latest commit etc.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
Checks are non-informative here, the last OS where the port builds is Catalina: https://ports.macports.org/port/aolserver/details/

**Comment by barracuda156** [comment]
I guess I can borrow @ryandesign fix from here: https://github.com/macports/macports-ports/commit/80d3dcac293a2a80c1a503c9a8f6e231ffd98001

**Comment by barracuda156** [comment]
@tobypeterson @mascguy Fixed now!

P. S. @kencu Sometimes I fix something for newer MacOS too :)

**Comment by barracuda156** [comment]
UPD. Perhaps solution should be somewhat different: https://github.com/aolserver/aolserver/pull/7#issuecomment-1237334761

Do not merge yet.

**Comment by mascguy** [comment]
> UPD. Perhaps solution should be somewhat different: [aolserver/aolserver#7 (comment)](https://github.com/aolserver/aolserver/pull/7#issuecomment-1237334761)
> 
> Do not merge yet.

Sounds good, changed status to Draft. (That's something you can do as well. And it's always a good idea to keep PRs as Draft, until you're ready for a final review.)

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]
In #11431 (which reverted 80d3dcac293), I remarked that Tcl software should be migrated to `Tcl_Panic()` instead of the system `panic()` (**edit**: I now notice this was already mentioned in the upstream PR), and that Tcl software tends to include system headers from a common header file (~~maybe nsd.h in this case~~ **edit: nsd.h already includes ctype.h**) rather than the individual source files.

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]
The configure script should also be patched (or possibly regenerated by newer autoconf, but patching is probably enough in this case):
```diff
--- configure.orig
+++ configure
@@ -2838,6 +2838,7 @@ _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
+#include <stdlib.h>
 #include <ctype.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
```

This fixes the issue where `checking for ANSI C header files…` outputs `no` on Xcode 12 and later, because its check is missing `#include <stdlib.h>`, causing `exit()` to be implicitly declared, which Xcode 12 and later consider an error by default:

```
configure:2866: clang -o conftest   conftest.c  >&5
configure:2860:7: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
      exit(2);
      ^
configure:2860:7: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
```

Submitted upstream: https://github.com/aolserver/aolserver/pull/8

**Comment by chrstphrchvz** [comment]
See https://github.com/aolserver/aolserver/pull/9 for alternative approach which migrates `panic()` usage to `Tcl_Panic()`.

Using this and the configure patch, along with the rest of @barracuda156’s  proposed changes to the portfile, I can successfully build the aolserver port on macOS 10.15.7 with Command Line Tools 12.4.

**Comment by barracuda156** [comment]
Oops, sorry, I will fix this now.

**Comment by barracuda156** [comment]
@reneeotten @pmetzger Could we merge this now? Maintainer timeout.

File: www/aolserver/Portfile
Comment (chrstphrchvz) [inline www/aolserver/Portfile:6]: The changes in patch-panic.diff have since been merged upstream; consider using the latest commit instead:

```suggestion
github.setup        aolserver aolserver 650ff5697c8b423f7cc338fbca64ac7f1d891e99
```

**Comment by barracuda156** [comment]
@chrstphrchvz Updated to the latest commit, dropped patch, rebased against master.

**Comment by barracuda156** [comment]
@reneeotten Are we good with this?

**Comment by pmetzger** [comment]
Merged. If there are remaining problems, we can fix them later.

**Comment by barracuda156** [comment]
@pmetzger Thank you!

**Comment by tobypeterson** [comment]
If someone else is interested in maintaining this port, that's fine with me. I happen to be the maintainer because I added the port 19(!) years ago.
```
commit c68a94ba1a8d1931a02eda488691dc936cb88d58
Author: Toby Peterson <toby@macports.org>
Date:   Fri Jan 16 23:44:56 2004 +0000
```

**Comment by barracuda156** [comment]
@tobypeterson Oh wow! That is a long time.
(I will probably not maintain it actively, so better someone else be a maintainer. Quite busy with math ports now.)

BTW, if you got a minute, could you take a look? https://github.com/macports/macports-ports/pull/17435

**Comment by pmetzger** [comment]
If Toby doesn't want to maintain it, why not set it nomaintainer?

**Pull request merged**
